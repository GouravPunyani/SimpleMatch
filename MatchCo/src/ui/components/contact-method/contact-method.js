// @flow

import React from 'react'
import { View, StyleSheet, Image, TouchableOpacity } from 'react-native'
import { Text } from '../..'
import { fontFamily, color, spacing } from '../../theme'

export type Props = {
  icon: string,
  text: string,
  onPress: () => *
};

const envelope = require('./envelope.png')
const phone = require('./phone.png')

export function ContactMethod (props: Props): React.Children {
  const icon = props.icon === 'phone' ? phone : envelope
  return (
    <TouchableOpacity style={styles.wrapper} onPress={props.onPress}>
      <View style={styles.imageWrap}>
        <Image style={styles[props.icon]} source={icon} resizeMode='contain' />
      </View>
      <Text style={styles.text}>{props.text}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
    paddingVertical: spacing.extraSmall
  },
  imageWrap: {
    minWidth: 35,
    paddingRight: spacing.extraSmall
  },
  envelope: { height: 23, width: 35 },
  phone: { height: 34, width: 34 },
  text: {
    paddingLeft: spacing.extraSmall,
    color: color.brand,
    fontFamily: fontFamily.nexaBold
  }
})
