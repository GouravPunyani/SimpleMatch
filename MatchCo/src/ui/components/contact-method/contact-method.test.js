// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { ContactMethod as Comp } from './contact-method'

test('renders correctly', () => {
  const tree = renderer.create(<Comp text='a' icon='b' onPress={() => null} />)
  expect(tree).toMatchSnapshot()
})
