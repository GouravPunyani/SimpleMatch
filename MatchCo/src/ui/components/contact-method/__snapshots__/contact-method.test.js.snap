// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "opacity": 1,
      "paddingVertical": 4,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "minWidth": 35,
        "paddingRight": 4,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={undefined}
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#333333",
          "fontFamily": "Nexa",
          "fontSize": 14,
        },
        Object {
          "color": "#E1CD96",
          "fontFamily": "NexaBold",
          "paddingLeft": 4,
        },
      ]
    }
  >
    a
  </Text>
</View>
`;
